src/common/csirf.cpp

```diff
@@ -136,7 +139,96 @@ void* ptr_z
 			objectFromHandle<DataContainer >(ptr_y);
 		DataContainer& z =
 			objectFromHandle<DataContainer >(ptr_z);
-		z.axpby(ptr_a, x, ptr_b, y);
+		z.xapyb(x, ptr_a, y, ptr_b);
+		return new DataHandle;
+	}
+	CATCH;
+}
+
+extern "C"
+void*
+cSIRF_xapyb_ss(
+const void* ptr_x, const void* ptr_a,
+const void* ptr_y, const void* ptr_b
+) {
+	try {
+		DataContainer& x =
+			objectFromHandle<DataContainer >(ptr_x);
+		DataContainer& y =
+			objectFromHandle<DataContainer >(ptr_y);
+		void* h = x.new_data_container_handle();
+		DataContainer& z = objectFromHandle<DataContainer>(h);
+		z.xapyb(x, ptr_a, y, ptr_b);
+		return h;
+	}
+	CATCH;
+}
+
+extern "C"
+void*
+cSIRF_xapyb_vv(
+const void* ptr_x, const void* ptr_a,
+const void* ptr_y, const void* ptr_b
+) {
+	try {
+		DataContainer& x =
+			objectFromHandle<DataContainer >(ptr_x);
+		DataContainer& y =
```

Comment:
these two functions have substantial overlap - why not have one `cSIRF_xapyb` with extra `const char* args` parameter that may have values `"ss"` or `"vv"` (and potentially other values, making it future proof)?

this would also solve lesser problem of breaking our naming scheme `cSIRF_nameOfFunctionInCamelCase`